### PR TITLE
Extend the Notifications API

### PR DESCRIPTION
**Pull request by flashd2n**
New Notifications methods:\

- list() -> returns all notifications saved in the session memory
- click() -> simulates a notification click
- clear() -> removes a notification from the state
- clearAll() -> removes all notifications from the state
- onRaised() -> notifies when a new notification was raised
- onClosed() -> notifies when a notification was closed or clicked and closed

New Platform Config options:
```
const config = {
    notifications: {
        enable: true,
        enableToasts: true,
    }
}
```

Allows to completely disable the notifications capabilities.
Allows to deny opening system toasts.

New notifications.raise options:
- severity?: "Low" | "Medium" | "High" | "Critical" | "None";
- showToast?: boolean;
- showInPanel?: boolean;

### CODE REVIEW

File: packages/web-platform/src/controllers/session.ts [hunk 1]
```diff
@@ -344,12 +343,18 @@ export class SessionStorageController {
         this.sessionStorage.setItem(this.extNotificationsNamespace, JSON.stringify(allNotifications));
     }
 
-    public getNotification(id: string): ExtensionNotification | undefined {
-        const allNotifications: ExtensionNotification[] = JSON.parse(this.sessionStorage.getItem(this.extNotificationsNamespace) as string);
+    public getNotification(id: string): Glue42Web.Notifications.NotificationData | undefined {
+        const allNotifications: Glue42Web.Notifications.NotificationData[] = JSON.parse(this.sessionStorage.getItem(this.extNotificationsNamespace) as string);
```
Comment (ksgeorgieva) [inline packages/web-platform/src/controllers/session.ts:347]: you may use this.getAllNotifications()

File: packages/web-worker/src/worker.ts [hunk 2]
```diff
@@ -249,4 +249,57 @@ export const setupCore: Glue42WebWorkerFactoryFunction = (config?: WebWorkerConf
...
+
+    self.addEventListener("notificationclose", (event: any) => {
+        let isPlatformOpen: boolean;
+
+        const channel = new BroadcastChannel(serviceWorkerBroadcastChannelName);
+
+        console.debug("Received a notification close event, checking if the platform is open");
+
+        const executionPromise = checkPlatformOpen()
+            .then((platformExists: boolean) => {
```
Comment (ksgeorgieva) [inline packages/web-worker/src/worker.ts:261]: not returning a promise => can combine current and previous callbacks

File: packages/web-platform/src/controllers/session.ts (hunk 1 above)
Comment (flashd2n) [reply packages/web-platform/src/controllers/session.ts:347]: fixed

File: packages/web-worker/src/worker.ts (hunk 2 above)
Comment (flashd2n) [reply packages/web-worker/src/worker.ts:261]: makes sense, fixed

File: e2e/tests/notifications/clearAll.spec.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        await glue.notifications.raise({ title: "asd" });
+
+        await glue.notifications.clearAll();
+
+        const allNotifications = await glue.notifications.list();
+
+        expect(allNotifications.length).to.eql(0);
+    });
+
+    it("should remove all notifications when two notifications was present - list check", async () => {
```
Comment (ksgeorgieva) [inline e2e/tests/notifications/clearAll.spec.js:38]: when two notifications were* present

**Review by ksgeorgieva — APPROVED** [review]

**Pull request merged**